### PR TITLE
Add str.merro60 pcb firmware

### PR DESCRIPTION
**Pull request by gaclee3b**
## Description

Adding firmware for a new custom 60% PCB based off previous merro60 PCB. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

n/a

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/chlx/str_merro60/config.h [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+#define MATRIX_COL_PINS { D0, D1, D2, E6, B0, B1, B2 }
+#define UNUSED_PINS
+// #define UNUSED_PINS { B4, B5, B6, D4, D5, D6, D7 }
+
+/* COL2ROW, ROW2COL*/
+#define DIODE_DIRECTION ROW2COL
+
+#define RGB_DI_PIN D3
+#ifdef RGB_DI_PIN
+#   define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/chlx/str_merro60/config.h:53]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/chlx/str_merro60/rules.mk [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+#
+BOOTMAGIC_ENABLE = yes      # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (drashna) [inline keyboards/chlx/str_merro60/rules.mk:18]:
```suggestion
NKRO_ENABLE = yes           # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
```

File: keyboards/chlx/str_merro60/str_merro60.c [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+along with this program.  If not, see <http://www.gnu.org/licenses/>. 
+*/ 
+
+#include "str_merro60.h"
+
+// Borrowed from hineybush h88
+void eeconfig_init_kb(void) {  // EEPROM is getting reset!
+  rgblight_enable(); // Enable RGB by default
+  rgblight_sethsv(0, 255, 128);  // Set default HSV - red hue, full saturation, medium brightness
+  rgblight_mode(RGBLIGHT_MODE_RAINBOW_SWIRL + 2); // set to RGB_RAINBOW_SWIRL by default
```
Comment (drashna) [inline keyboards/chlx/str_merro60/str_merro60.c:24]: These are not needed, as there are defines to handle this. 

https://docs.qmk.fm/#/feature_rgblight?id=configuration

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by gaclee3b — COMMENTED** [review]
update config.h, rules.mk per drashna recs

File: keyboards/chlx/str_merro60/str_merro60.c (hunk 3 above)
Comment (gaclee3b) [reply keyboards/chlx/str_merro60/str_merro60.c]: removed eeprom reset code in *.cadded RGBLIGHT_DEFAULT_MODE and specified RGBLIGHT_EFFECT_* in place of RGBLIGHT_ANIMATIONS in

File: keyboards/chlx/str_merro60/info.json [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+    "url": "www.github.com/gaclee3b",
+    "maintainer": "gaclee3b",
+    "layout_aliases": {
+        "LAYOUT_default": "LAYOUT_all",
+        "LAYOUT_hhkb": "LAYOUT_60_hhkb",
+        "LAYOUT_iso": "LAYOUT_60_iso_split_bs_rshift",
+        "LAYOUT_tsangan": "LAYOUT_60_tsangan_hhkb"
+    },
+    "width": 15,
+    "height": 5,
```
Comment (drashna) [inline keyboards/chlx/str_merro60/info.json:12]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/chlx/str_merro60/rules.mk [hunk 5]
```diff
@@ -0,0 +1,20 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes      # Virtual DIP switch configuration
```
Comment (fauxpark) [inline keyboards/chlx/str_merro60/rules.mk:10]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/chlx/str_merro60/readme.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
+# str.merro60
+
+str.merro60 PCB
+
+A standard 60% format keyboard pcb with USB-C and unified daughterboard compatibility.
+Has modified cutouts to allow usage of certain tray mount cases into gummy gasket o-ring mount.
+Has RGB underglow.
+
+Keyboard Maintainer: [Alexander Lee](https://github.com/gaclee3b)
+Hardware Supported: chlx PCB str.merro60.1.1
```
Comment (fauxpark) [inline keyboards/chlx/str_merro60/readme.md:10]:
```suggestion
* Keyboard Maintainer: [Alexander Lee](https://github.com/gaclee3b)
* Hardware Supported: chlx PCB str.merro60.1.1
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by gaclee3b** [comment]
do I need to do anything else / make any other changes at this point?

**Comment by drashna** [comment]
> do I need to do anything else / make any other changes at this point?

Mostly, was waiting for breaking changes cycle to finish.  Master was locked until the 28th.

**Pull request merged**